addons/stock/models/stock_package_level.py

```diff
@@ -54,9 +55,14 @@ def _set_is_done(self):
                 if not package_level.is_fresh_package:
                     ml_update_dict = defaultdict(float)
                     for quant in package_level.package_id.quant_ids:
-                        corresponding_ml = package_level.move_line_ids.filtered(lambda ml: ml.product_id == quant.product_id and ml.lot_id == quant.lot_id)
-                        if corresponding_ml:
-                            ml_update_dict[corresponding_ml[0]] += quant.quantity
+                        corresponding_mls = package_level.move_line_ids.filtered(lambda ml: ml.product_id == quant.product_id and ml.lot_id == quant.lot_id)
+                        to_dispatch = quant.quantity
+                        if corresponding_mls:
+                            while float_compare(to_dispatch, 0, precision_digits=2) and corresponding_mls:
```

Comment:
`float_compare(to_dispatch, 0, precision_digits=2)` --> `not float_is_zero(to_dispatch, precision_rounding=quant.product_uom_id.rounding)`

maybe use the precision rounding of the uom (and not 2 by default)